resilience4j-retry/src/main/java/io/github/resilience4j/retry/internal/RetryImpl.java

```diff
@@ -177,6 +183,9 @@ public void onComplete() {
         public boolean onResult(T result) {
             if (null != resultPredicate && resultPredicate.test(result)) {
                 int currentNumOfAttempts = numOfAttempts.incrementAndGet();
+                if(postActionOnRetryBiConsumer != null){
+                    postActionOnRetryBiConsumer.accept(currentNumOfAttempts, result);
```

Comment:
What should happen if the consumer itself throws an exception?
I think we should catch the exception and log it, but not rethrow it.

Reply:
Shouldn't we execute the consumer only when a retry attempt needs to be performed? I think the consumer must not be invoked when no retry attempt must be performed, or? In that case the normal code flow gets the result and can do whatever is necessary.
We could move this line of code in front of  `waitIntervalAfterFailure(currentNumOfAttempts, Either.right(result));`, or?
We could also rename `postActionOnRetryBiConsumer` to `consumeResultBeforeRetryAttempt`. Wdyt?

Reply:
I though possible exception in the postAction should be handled by the creator of the function. If we catch the exception and just log it, it might be unrecognized. The behavior is at the moment equal to the resultPredicate test isn't it?

I agree with you. We should not perform the postAction if no retry is needed and the renaming to "consumeResultBeforeRetryAttempt" would clarify this behavior. However the postAction is currently only performed if the resultPredicated returned true e.g. we already have the desired behavior don't we?

Reply:
You're right if we rename the consumer to consumeResultBeforeRetryAttempt we should not invoke the consumer after the last retry. The user can do whatever he wants with the returned object in the normal code-flow. I will move the consumer accordingly.